### PR TITLE
Meta: Extract AK into its own library on Lagom

### PR DESCRIPTION
**Pull request by trflynn89**
We currently bundle AK with LibCore on Lagom. This means that to use AK, all libraries must also depend on LibCore. This will create circular dependencies when we create LibURL, as LibURL will depend on LibUnicode, which will depend on LibCore, which will depend on LibURL.

Then as a follow up, remove LibCore as a dependency from libraries that are not using it.

Ref #22884

### CODE REVIEW

**Comment by ADKaster** [comment]
In the GN build, "//AK:sources" is a dependency of LibCore.

**Comment by trflynn89** [comment]
> In the GN build, "//AK:sources" is a dependency of LibCore.

The GN build seems far enough out-of-date that I was going to tackle that separately

**Comment by trflynn89** [comment]
Was easy enough to get caught back up, so I included GN here

File: Meta/gn/secondary/AK/BUILD.gn [hunk 1]
```diff
@@ -11,16 +11,18 @@ source_set("AK") {
   public_configs = [ ":ak_headers" ]
   public_deps = [ ":ak_debug_gen" ]
 
-  # NOTE: Headers only!
+  # FIXME: Split out non-kernel sources to their own set
```
Comment (ADKaster) [inline Meta/gn/secondary/AK/BUILD.gn:14]: AK is still a source set here, which means everyone will get the object files linked in. Probably not quite what we need.

**Review by ADKaster — COMMENTED** [review]
Lgtm, modulo AK being an OBJECT library in the gn build

File: Meta/gn/secondary/AK/BUILD.gn (hunk 1 above)
Comment (trflynn89) [reply Meta/gn/secondary/AK/BUILD.gn:16]: Ah good catch. I'm guessing we'll have to play with it once Serenity is building with GN, but I've made it a shared lib now.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**